js/exercise_one/index.js

```diff
@@ -0,0 +1,90 @@
+function sortTableByCost(table) {
+
+  const tBody = table.tBodies[0];
+  const rows = Array.from(tBody.querySelectorAll("tr"));
+
+  const sortedRows = rows.sort((a, b) => {
+    const aCost = parseFloat(a.querySelector(`td:nth-child(${ 1 + 1})`).textContent.substring(1));
+    const bCost = parseFloat(b.querySelector(`td:nth-child(${ 1 + 1})`).textContent.substring(1));
+    return aCost >= bCost ? 1 : -1;
+  });
+
+  while(tBody.firstChild){ // ACA SE TRANCA
+    tBody.removeChild(tBody.firstChild);
+  }
+  tBody.append(...sortedRows);
+}
+
+function chackAffordableItems(budget, table) {
+
+  const rows = table.getElementsByTagName('tr');
+  for (let i = 1; i < rows.length; i++) {
+    const row = rows[i];
+    const cells = row.getElementsByTagName('td');
+
+    if (cells.length >= 2) {
+      const cost = parseFloat(cells[1].textContent.substring(1));
+      const checkbox = cells[2].querySelector('input[type="checkbox"]');
+
+      if (cost <= budget) {
+        checkbox.checked = true;
+        budget -= cost;
+        console.log("entro");
+      } else {
+        checkbox.checked = false;
+        break;
+      }
+    }
+  }
+}
+
+//  Optimize list
+function optimize() {
+
+  const budget = parseFloat(document.getElementById('budget').value);
+  document.getElementById('title-budget').innerHTML = budget;
+
+  if (isNaN(budget)) {
+    // Do nothing
+    return;
+  }
```

Comment:
You can write this in one line
```suggestion
  if (isNaN(budget)) return;
```